pkg/api/ldap_debug.go

```diff
@@ -278,13 +278,15 @@ func (server *HTTPServer) GetUserFromLDAP(c *models.ReqContext) Response {
 
 	orgRoles := []LDAPRoleDTO{}
 
-	// First, let's find the groupDN that we did match by inspecting the assigned user OrgRoles.
```

Comment:
I think we need to match all three conditions - there is a world where you can match the group but not the org/role.

The LDAP code is structured in a way that if you have a role assigned for an org it won't match any more beyond the first. The order of how the roles are specified in the TOML file matter.

Reply:
Thanks for pointing this out, I wasn't aware of that. I've updated the code to reflect this.